pkg/config/config.go

```diff
@@ -21,45 +21,46 @@ const defaultConfigFile = "athens.toml"
+	ProtocolWorkers   int           `validate:"required" envconfig:"ATHENS_PROTOCOL_WORKERS" json:"protocolWorkers"`
+	LogLevel          string        `validate:"required" envconfig:"ATHENS_LOG_LEVEL" json:"logLevel"`
+	CloudRuntime      string        `validate:"required" envconfig:"ATHENS_CLOUD_RUNTIME" json:"cloudRuntime"`
+	EnablePprof       bool          `envconfig:"ATHENS_ENABLE_PPROF" json:"enablePprof"`
+	PprofPort         string        `envconfig:"ATHENS_PPROF_PORT" json:"pprofPort"`
+	FilterFile        string        `envconfig:"ATHENS_FILTER_FILE" json:"filterFile"`
+	TraceExporterURL  string        `envconfig:"ATHENS_TRACE_EXPORTER_URL" json:"traceExporterUrl"`
+	TraceExporter     string        `envconfig:"ATHENS_TRACE_EXPORTER" json:"traceExporter"`
+	StatsExporter     string        `envconfig:"ATHENS_STATS_EXPORTER" json:"statsExporter"`
+	StorageType       string        `validate:"required" envconfig:"ATHENS_STORAGE_TYPE" json:"storageType"`
+	GlobalEndpoint    string        `envconfig:"ATHENS_GLOBAL_ENDPOINT" json:"globalEndpoint"` // This feature is not yet implemented
+	Port              string        `envconfig:"ATHENS_PORT" json:"athensPort"`
+	BasicAuthUser     string        `envconfig:"BASIC_AUTH_USER" json:"basicAuthUser"`
+	BasicAuthPass     string        `envconfig:"BASIC_AUTH_PASS" json:"basicAuthPass"`
+	PropagateAuthHost string        `envconfig:"ATHENS_PROPAGATE_AUTH_HOST" json:"propagateAuthHost"`
+	ForceSSL          bool          `envconfig:"PROXY_FORCE_SSL" json:"forceSsl"`
+	ValidatorHook     string        `envconfig:"ATHENS_PROXY_VALIDATOR" json:"validatorHook"`
+	PathPrefix        string        `envconfig:"ATHENS_PATH_PREFIX" json:"pathPrefix"`
+	NETRCPath         string        `envconfig:"ATHENS_NETRC_PATH" json:"netrcPath"`
+	GithubToken       string        `envconfig:"ATHENS_GITHUB_TOKEN" json:"githubToken"`
+	HGRCPath          string        `envconfig:"ATHENS_HGRC_PATH" json:"hgrcPath"`
+	TLSCertFile       string        `envconfig:"ATHENS_TLSCERT_FILE" json:"tlsCertFile"`
+	TLSKeyFile        string        `envconfig:"ATHENS_TLSKEY_FILE" json:"tlsKeyFile"`
+	SumDBs            []string      `envconfig:"ATHENS_SUM_DBS" json:"sumDbs"`
+	NoSumPatterns     []string      `envconfig:"ATHENS_GONOSUM_PATTERNS" json:"noSumPatterns"`
+	DownloadMode      mode.Mode     `envconfig:"ATHENS_DOWNLOAD_MODE" json:"downloadMode"`
+	DownloadURL       string        `envconfig:"ATHENS_DOWNLOAD_URL" json:"downloadUrl"`
+	SingleFlightType  string        `envconfig:"ATHENS_SINGLE_FLIGHT_TYPE" json:"singleFlightType"`
+	RobotsFile        string        `envconfig:"ATHENS_ROBOTS_FILE" json:"robotsFile"`
+	IndexType         string        `envconfig:"ATHENS_INDEX_TYPE" json:"indexType"`
+	SingleFlight      *SingleFlight `json:"singleFlight"`
+	Storage           *Storage      `json:"storage"`
+	Index             *Index        `json:"index"`
+	ConfigPort        *string       `envconfig:"ATHENS_CONFIG_PORT" json:"configPort"`
```

Comment:
No need to make this a pointer to a string, we can just make it a string and check in main.go for `!= ""` instead of `!= nil`. 

It's pretty common and we do it a lot for the other config fields :v: